src/python/src/pyconfigoptions.cpp

```diff
@@ -0,0 +1,289 @@
+{
+    try {
+        auto cfgOpts = ConfigOptions::create_from_file(filename);
+        ConfigOptions* pCfgOpts = cfgOpts.release();
+        return new PyConfigOptions(pCfgOpts);
+    }
+    catch (Exception& e) {
+        // exception is already prepared for caller
+        throw;
+    }
+    catch (std::exception& e) {
+        // should never happen
+        throw SRInternalException(e.what());
+    }
+    catch (...) {
+        // should never happen
+        throw SRInternalException("A non-standard exception was encountered "\
+                                  "while executing create_from_file.");
+    }
+}
+
+
+// Instantiate ConfigOptions from a string containing a JSON blob
+PyConfigOptions* PyConfigOptions::create_from_string(
+    const std::string& json_blob)
+{
+    try {
+        auto cfgOpts = ConfigOptions::create_from_string(json_blob);
+        ConfigOptions* pCfgOpts = cfgOpts.release();
+        return new PyConfigOptions(pCfgOpts);
+    }
+    catch (Exception& e) {
+        // exception is already prepared for caller
+        throw;
+    }
+    catch (std::exception& e) {
+        // should never happen
+        throw SRInternalException(e.what());
+    }
+    catch (...) {
+        // should never happen
+        throw SRInternalException("A non-standard exception was encountered "\
+                                  "while executing create_from_string.");
+    }
+}
+#endif
+
+// Retrieve the value of a numeric configuration option
+// from the selected source
+int64_t PyConfigOptions::get_integer_option(const std::string& key)
+{
+    try {
+        if (_configoptions == NULL) {
+            throw SRRuntimeException(
+                "Attempted to call get_integer_option "\
+                "from a non-factory constructed ConfigOptions");
+        }
+        return _configoptions->get_integer_option(key);
+    }
+    catch (Exception& e) {
+        // exception is already prepared for caller
+        throw;
+    }
+    catch (std::exception& e) {
+        // should never happen
+        throw SRInternalException(e.what());
+    }
+    catch (...) {
+        // should never happen
+        throw SRInternalException("A non-standard exception was encountered "\
+                                  "while executing get_integer_option.");
+    }
+}
+
+// Retrieve the value of a string configuration option
+// from the selected source
+std::string PyConfigOptions::get_string_option(const std::string& key)
+{
+    try {
+        if (_configoptions == NULL) {
+            throw SRRuntimeException(
+                "Attempted to call get_string_option "\
+                "from a non-factory constructed ConfigOptions");
+        }
+        return _configoptions->get_string_option(key);
+    }
+    catch (Exception& e) {
+        // exception is already prepared for caller
+        throw;
+    }
+    catch (std::exception& e) {
+        // should never happen
+        throw SRInternalException(e.what());
+    }
+    catch (...) {
```

Comment:
Out of scope for this ticket, so don't feel the need to handle here, but this seems like a another try/catch/catch/catch that we can template-ize/decorate. Maybe open a new ticket?

Reply:
done